src/backend/dx11/src/lib.rs

```diff
@@ -120,7 +133,7 @@ impl hal::Instance for Instance {
                         heap_index: 0,
                     },
                     hal::MemoryType {
-                        properties: Properties::DEVICE_LOCAL | Properties::CPU_VISIBLE | Properties::CPU_CACHED,
+                        properties: Properties::DEVICE_LOCAL, //| Properties::CPU_VISIBLE | Properties::CPU_CACHED,
```

Comment:
this is fully matching the first type now?

Reply:
Still messing around with the memory types a bit. Comment was to make it implicitly ignore a certain heap IIRC (not the best way I know).

For the memory types I think it's better to start conservatively then increase support afterwards.

Right now I have:
* `DEVICE_LOCAL` which works as intended (maps to `USAGE_DEFAULT` and should work for both buffers and images)
* `DEVICE_LOCAL | CPU_VISIBLE | CPU_CACHED` which is a bit more flakey. This is backed by one resource with `USAGE_STAGING` and another one with `USAGE_DEFAULT`. Reasoning here is that only `STAGING` resources in dx11 have CPU read & write, while at the same time can't be bound to any stage in the pipeline (hence another resource for device local memory)

I'm not entirely sure how solid the last method is, but I'm not too worried about this area.

Reply:
Please check out the heap mapping in Metal: https://github.com/gfx-rs/gfx/blob/447a0da3ad43310b124e4632a532b4fcba1f27a8/src/backend/metal/src/device.rs#L138

It's not ideal, but at least there isn't as much bit twiddling going on in the user code. Maybe we could have those memory flags in D3D11 as well? Not exactly the same, of course, just in terms of organization.

Reply:
I don't see why we'd want to provide `DEVICE_LOCAL` *and* `CPU_VISIBLE` if D3D11 doesn't have that locally. Can we just expose `CPU_VISIBLE` for staging resources and leave the copy to/from device local memory to the user?

Reply:
Problem is `USAGE_STAGING` only allows for CopyResource/CopySubresourceRegion, which requires resources to be of the same type, so no buffer -> image copies for example